### PR TITLE
add django 3.2 and 4.0 support

### PR DESCRIPTION
**Pull request by umarmughal824**
Add django 3.2 and 4.0 support

I am facing the following warning while upgrading the django to 3.2 which take the default variable for `DEFAULT_AUTO_FIELD` [ref](https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field)

```
web_1     | robots.Rule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
web_1     | 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
web_1     | robots.Url: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
web_1     | 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

So in lieu of that I am facing the error in my CI/CD that is

```
[TEST SUITE]./scripts/test/detect_missing_migrations.sh
Error: one or more migrations are missing
Migrations for 'robots': /opt/hostedtoolcache/Python/3.7.5/x64/lib/python3.7/site-packages/robots/migrations/0002_auto_20210923_1203.py - Alter field id on rule - Alter field id on url
```

### CODE REVIEW

**Comment by umarmughal824** [comment]
@jezdez could you please take a look at that PR?

File: tox.ini [hunk 1]
```diff
@@ -2,7 +2,7 @@
 envlist =
     # list of supported Django/Python versions:
     # https://docs.djangoproject.com/en/3.1/faq/install/#what-python-version-can-i-use-with-django
-    py{36,37,38,39}-dj{22,30,31}
+    py{36,37,38,39}-dj{22,30,31,32}
     py{38,39}-djmain
```
Comment (smithdc1) [inline tox.ini:6]: Let's add Django 4.0 support as well while we're here?

File: tox.ini [hunk 2]
```diff
@@ -27,6 +27,7 @@ deps =
     dj22: django>=2.2,<2.3
     dj30: django>=3.0,<3.1
```
Comment (smithdc1) [inline tox.ini:28]: We can also drop Django 3.0 support.

Comment (umarmughal824) [reply tox.ini:28]: sure

File: tox.ini (hunk 1 above)
Comment (umarmughal824) [reply tox.ini:6]: Django 4.0 is not yet final so better to wait for the final release which is due in December.
https://code.djangoproject.com/wiki/Version4.0Roadmap

**Comment by umarmughal824** [comment]
@smithdc1 your feedback is addressed so it's ready for review again.

File: tox.ini (hunk 1 above)
Comment (smithdc1) [reply tox.ini:6]: It's final in terms of features and changes. The only patches being applied to the 4.0 branch are bug fixes. Folk are adding the classifier to their packages already. See:

https://pypi.org/search/?c=Framework+%3A%3A+Django+%3A%3A+4.0

**Comment by umarmughal824** [comment]
@smithdc1 I have added 4.0 support as well and it seems to be working as well.

File: tox.ini (hunk 1 above)
Comment (umarmughal824) [reply tox.ini:6]: @smithdc1 I'm waiting for your review, thanks

File: README.rst [hunk 3]
```diff
@@ -22,9 +22,9 @@ directory or on ReadTheDocs: https://django-robots.readthedocs.io/
 
 Supported Django versions
 -------------------------
-
```
Comment (smithdc1) [inline README.rst:25]: Please re-add the blank line.

File: tox.ini [hunk 4]
```diff
@@ -2,7 +2,7 @@
 envlist =
     # list of supported Django/Python versions:
     # https://docs.djangoproject.com/en/3.1/faq/install/#what-python-version-can-i-use-with-django
-    py{36,37,38,39}-dj{22,30,31}
+    py{36,37,38,39}-dj{22,31,32,40}
```
Comment (smithdc1) [inline tox.ini:5]:
```suggestion
    py{36,37,38,39}-dj{22,31,32}
    py{38,39}-dj{40}
```

**Review by smithdc1 — COMMENTED** [review]
Thanks for this 👍 

I think we're nearly there. I've re-run the tests but expecting to see them fail as Django 4.0 should fail with Python 3.6&3.7.

**Comment by smithdc1** [comment]
@umarmughal824 -- Sorry, having a bad day. Thanks for the contribution. 🙏

**Comment by tony** [comment]
@umarmughal824 This is prereleased at [5.0b1](https://pypi.org/project/django-robots/5.0b1/) if you'd like to give it a go.  Would like some testing before final is published

**Comment by tony** [comment]
@umarmughal824 and @smithdc1 There is an issue that came up due to this PR in #124, can you take a closer look?

**Pull request merged**